Sources/PackageLoading/ManifestJSONParser.swift

```diff
@@ -92,78 +123,36 @@ enum ManifestJSONParser {
     }
 
     private static func parseDependency(
-        json: JSON,
+        dependency: Serialization.PackageDependency,
         toolsVersion: ToolsVersion,
         packageKind: PackageReference.Kind,
         identityResolver: IdentityResolver,
         fileSystem: TSCBasic.FileSystem
     ) throws -> PackageDependency {
-        if let kindJSON = try? json.getJSON("kind") {
-            // new format introduced 7/2021
-            let type: String = try kindJSON.get("type")
-            if type == "fileSystem" {
-                let name: String? = kindJSON.get("name")
-                let path: String = try kindJSON.get("path")
-                return try Self.parseFileSystemDependency(
-                    packageKind: packageKind,
-                    at: path,
-                    name: name,
-                    identityResolver: identityResolver,
-                    fileSystem: fileSystem
-                )
-            } else if type == "sourceControl" {
-                let name: String? = kindJSON.get("name")
-                let location: String = try kindJSON.get("location")
-                let requirementJSON: JSON = try kindJSON.get("requirement")
-                let requirement = try PackageDependency.SourceControl.Requirement(v4: requirementJSON)
-                return try Self.parseSourceControlDependency(
-                    packageKind: packageKind,
-                    at: location,
-                    name: name,
-                    requirement: requirement,
-                    identityResolver: identityResolver,
-                    fileSystem: fileSystem
-                )
-            } else if type == "registry" {
-                let identity: String = try kindJSON.get("identity")
-                let requirementJSON: JSON = try kindJSON.get("requirement")
-                let requirement = try PackageDependency.Registry.Requirement(v4: requirementJSON)
-                return try Self.parseRegistryDependency(
-                    identity: .plain(identity),
-                    requirement: requirement,
-                    identityResolver: identityResolver
-                )
-            } else {
-                throw InternalError("Unknown dependency type \(kindJSON)")
-            }
-        } else {
-            // old format, deprecated 7/2021 but may be stored in caches, etc
-            let name: String? = json.get("name")
-            let url: String = try json.get("url")
-
-            // backwards compatibility 2/2021
```

Comment:
did we preserve all this logic?

Reply:
No, the backwards compatibility is gone here, that's the only way to get away from hand-coding. I think that's OK since we should simply repopulate the cache.

Reply:
okay if we make sure the cache is invalidated

Reply:
Yep, that seems to work fine, but I encountered an interesting case while testing. 

If we load the wrong PD library (e.g. installed instead of inferior), we'll get a pretty confusing error, so might be worth handling explicitly, e.g. by adding a version into the JSON. That'll make the current error more obvious and can also be used to handle this type of mismatch more gracefully in the future.

Reply:
(also separately we should look into why running `swift-build` via a relative path breaks the way it picks up the inferior PD libs and falls back to using the installed version)

Reply:
Update to add versioning. The error when using a pre-5.9 PD library will be a bit ugly, but I think it's important to retain the underlying error since we can't definitively know what happened:

```
error: 'swift-package-manager': serialized JSON uses unsupported version 1, indicating use of a mismatched PackageDescription library, underlying error: keyNotFound(CodingKeys(stringValue: "version", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: \"version\", intValue: nil) (\"version\").", underlyingError: nil))
```